test/launcher.spec.js

```diff
@@ -239,6 +240,23 @@ module.exports.addTests = function({testRunner, expect, defaultBrowserOptions, p
         expect(page.url()).toBe(server.EMPTY_PAGE);
         await browser.close();
       });
+      it('should pass the timeout parameter to browser.waitForTarget', async() => {
+        const expectedTimeout = 100000;
+        const options = Object.assign({}, defaultBrowserOptions, {
+          timeout: expectedTimeout
+        });
+        const originalProto = Browser.prototype.waitForTarget;
+        let resTimeout = 0;
+        // function() because we want Browser's 'this'
+        Browser.prototype.waitForTarget = async function(predicate, options) {
```

Comment:
The pattern we use to test timeout is setting the timeout to 1 and expecting to fail. [See](https://github.com/puppeteer/puppeteer/blob/eddb23b52175a3dad8cec4a5e9b3c13e58f6f144/test/page.spec.js#L501)